realm/realm-library/src/androidTest/java/io/realm/OrderedCollectionChangeSetTests.java

```diff
@@ -465,4 +479,78 @@ public void onChange(RealmResults<Dog> collection, @Nullable OrderedCollectionCh
             }
         });
     }
+
+    // To reproduce https://github.com/realm/realm-java/issues/5507
+    // 1. Add listener to a collection
+    // A. change the collection in a background thread
+    // 2. Remove the listener
+    // 3. Add another listener
+    // 4. the listener added in step 3 should be triggered with change set in step A
+    @Test
+    @RunTestInLooperThread
+    public void addChangeListener_bug5507() throws InterruptedException {
+        // FIXME: See https://github.com/realm/realm-object-store/issues/605
+        if (type == ObservablesType.REALM_RESULTS) looperThread.testComplete();
+
+        Realm realm = looperThread.getRealm();
+        populateData(realm, 1);
+
+        OrderedRealmCollectionChangeListener<OrderedRealmCollection<Dog>> listener1 =
+                new OrderedRealmCollectionChangeListener<OrderedRealmCollection<Dog>>() {
+            @Override
+            public void onChange(OrderedRealmCollection<Dog> dogs, @Nullable OrderedCollectionChangeSet changeSet) {
+                fail();
+            }
+        };
+
+        OrderedRealmCollection<Dog> dogs = getTestingCollection(realm);
+        assertEquals(1, dogs.size());
+
+        if (type == ObservablesType.REALM_LIST) {
+            //noinspection unchecked
+            ((RealmList) dogs).addChangeListener(listener1);
+        } else {
+            //noinspection unchecked
+            ((RealmResults) dogs).addChangeListener(listener1);
```

Comment:
Is this `else` statement ever reached?

Reply:
Or are you just keeping them here so the test case is ready for the FIXME?

Reply:
yeah, make it ready for the FIXME. Added return statement at the beginning.